week-2/Homework/mandatory/3-playing-computer.js

```diff
@@ -5,38 +5,50 @@
   during each step of the program execution.
 
   Answer the following questions:
-
-  1. This program throws an error. Why? (If you can't find it, try executing it).
+  
+task 1 
+  1. This program throws an error. Why? (If you can't find it, try executing it).  //because "b" is not defined because it had no varruble with the value./
   2. Remove the line that throws the error.
-  3. What is printed to the console?
-  4. How many times is "f1" called?
-  5. How many times is "f2" called?
-  6. What value does the "a" parameter take in the first "f1" call?
+
+task 2
+  3. What is printed to the console?  // 2, 6, 4, 9, 6, 13, 8   
+  console.log(x);   2
+  console.log(a);   6
+  console.log(d);   4, 6, 8 
+  console.log(e);   9, 13
+
+  4. How many times is "f1" called?   // 1st call of f1 is when consol.log(e) = 9,  //  last time f1 is called and resulting in console.log(e) = 13  
+  5. How many times is "f2" called?  // 1st call of f2 is when console.log(d) = 4, second time f2 is called and resulting in console.log(d) = 6   // last time f2 is called and resulting in console.log(d) = 8
+  6. What value does the "a" parameter take in the first "f1" call?        // value 8 
```

Comment:
- the first call for `f1` when `i =1`, 
- we are calling the function by passing `f1( i , a)`
- then parameter `a` takes `i = 1`.
- parameter `b` takes `a=6`.

Reply:
execute this program here http://pythontutor.com/visualize.html#mode=edit
and see how things are going step by step